pkg/controller/sync/controller.go

```diff
@@ -614,16 +642,33 @@ func updatePropagatedVersion(adapter federatedtypes.FederatedTypeAdapter, versio
 
 	if version == nil {
 		version := newVersion(versions, templateMeta, adapter.Target().Kind(), overrideVersion)
-		_, err := adapter.FedClient().FederationV1alpha1().PropagatedVersions(version.Namespace).Create(version)
+
+		var err error
+		if isNamespaceKind(adapter.Template().Kind()) {
```

Comment:
This conditional is redundant given that the namespace for ``version`` is already correct.